### PR TITLE
feat: support custom agent identity prefixes

### PR DESCRIPTION
**Pull request by smolpaws**
## Summary
- add an AgentContext system-message prefix seam in the SDK
- let Agent replace the stock identity line while keeping the upstream operational prompt body intact
- cover the new prefix behavior with focused AgentContext and Agent system-prompt tests

## Validation
- npx vitest run packages/agent-sdk/src/sdk/context/__tests__/agent-context.test.ts packages/agent-sdk/src/sdk/runtime/__tests__/Agent.system-prompt.test.ts --exclude='.git/**'
- npm pack

---

<a href="https://app.devin.ai/review/enyst/openhands-tab/pull/1010" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

## Summary by CodeRabbit

* **New Features**
  * Added optional system message prefix configuration to customize how agents identify themselves in system prompts.
  * System message prefixes can be dynamically updated during agent runtime and are applied to all subsequent LLM requests.
  * Empty or unset prefixes automatically revert to default behavior.

### Review
- Gemini flagged one missing edge-case test for empty/whitespace prefixes; added the coverage and resolved the inline thread.
- Mandatory roasted review was recorded via Agent Mail (`PinkStone`): initial concern was that the first implementation removed the stock identity line with brittle string replacement inside `Agent.buildSystemPrompt()`.
- Follow-up commit `0b12b0e` addressed that by composing the system prompt structurally as `identity + body` via a dedicated `SYSTEM_PROMPT_BODY` constant; reviewer result: **APPROVED after latest structural follow-up**.
- Devin review reported no issues.
- CodeRabbit never produced actionable inline feedback on this PR after the final push; proceeded once all required checks and review-thread gates were green per repo policy.

### CODE REVIEW

File: packages/agent-sdk/src/sdk/context/__tests__/agent-context.test.ts [hunk 1]
```diff
@@ -314,6 +317,18 @@ describe('AgentContext', () => {
...
+  describe('getSystemMessagePrefix', () => {
+    it('returns null when no prefix is configured', () => {
+      const context = new AgentContext();
+      expect(context.getSystemMessagePrefix()).toBeNull();
+    });
+
+    it('returns the trimmed prefix when configured', () => {
+      const context = new AgentContext({ systemMessagePrefix: '  You are smolpaws.  ' });
+      expect(context.getSystemMessagePrefix()).toBe('You are smolpaws.');
+    });
```
[the remark this answers is not in the document]
Comment (smolpaws) [reply packages/agent-sdk/src/sdk/context/__tests__/agent-context.test.ts:334]: Added the empty/whitespace prefix coverage in the latest push.
